### PR TITLE
add bf dialog command and unit tests

### PR DESCRIPTION
**Pull request by tomlm**
* **bf dialog:merge** --> merges schema files
* **bf dialog:verify** --> verifies that .dialog files are correct

### CODE REVIEW

File: packages/dialog/src/commands/dialog/verify.ts [hunk 1]
```diff
@@ -1,25 +1,131 @@
...
+
+    consoleError(msg: string): void {
+        this.errors++;
+        console.error(chalk.default.redBright(`${this.currentFile}: error: ${msg}`));
     }
-  }
 }
+
+
+// interface IPackage {
```
Comment (carlosscastro) [inline packages/dialog/src/commands/dialog/verify.ts]: remove unused interface

File: packages/dialog/README.md [hunk 2]
```diff
@@ -30,7 +30,6 @@ USAGE
 <!-- commands -->
 * [`oclif-example dialog:merge [FILE]`](#oclif-example-dialogmerge-file)
```
Comment (munozemilio) [inline packages/dialog/README.md]: Can you please update readme file?
npm script prepack does it automatically

File: packages/dialog/package.json [hunk 3]
```diff
@@ -4,23 +4,35 @@
   "author": "Microsoft",
   "bugs": "https://github.com/microsoft/botframework-cli/issues",
   "dependencies": {
-    "@oclif/command": "~1.5.13",
+    "@oclif/command": "^1.5.18",
```
Comment (munozemilio) [inline packages/dialog/package.json]: use ~ instead of ^

File: packages/dialog/package.json [hunk 4]
```diff
@@ -40,16 +52,16 @@
...
-    "bin": "oclif-example",
+    "bin": "bf",
     "devPlugins": [
       "@oclif/plugin-help"
     ]
   },
   "repository": "microsoft/botframework-cli",
   "scripts": {
     "postpack": "rimraf oclif.manifest.json",
-    "posttest": "tslint -p test -t stylish",
```
Comment (munozemilio) [inline packages/dialog/package.json]: Why removing this?

File: packages/dialog/src/commands/dialog/verify.ts [hunk 5]
```diff
@@ -1,25 +1,131 @@
...
+                this.error(`${this.errors} found.`);
+            }
+            else {
+                this.log(`${this.errors} found.`);
+            }
+        }
+    }
+
+    consoleMsg(msg: string): void {
+        console.log(chalk.default(msg));
```
Comment (munozemilio) [inline packages/dialog/src/commands/dialog/verify.ts]: please use this.log()

File: packages/dialog/src/commands/dialog/verify.ts [hunk 6]
```diff
@@ -1,25 +1,131 @@
...
+    }
+
+    consoleLog(msg: string): void {
+        console.log(chalk.default.gray(msg));
+    }
+
+    consoleWarn(msg: string): void {
+        this.warnings++;
+        console.warn(chalk.default.yellowBright(`${this.currentFile}: warning: ${msg}`));
```
Comment (munozemilio) [inline packages/dialog/src/commands/dialog/verify.ts]: please use this.warn()

File: packages/dialog/src/commands/dialog/verify.ts [hunk 7]
```diff
@@ -1,25 +1,131 @@
...
+    }
+
+    consoleError(msg: string): void {
+        this.errors++;
+        console.error(chalk.default.redBright(`${this.currentFile}: error: ${msg}`));
```
Comment (munozemilio) [inline packages/dialog/src/commands/dialog/verify.ts]: this.err()

File: packages/dialog/src/commands/dialog/verify.ts [hunk 8]
```diff
@@ -1,25 +1,131 @@
-import {Command, flags} from '@oclif/command'
+import { Command, flags } from '@oclif/command';
```
Comment (munozemilio) [inline packages/dialog/src/commands/dialog/verify.ts]: Please use bf-cli-command to get telemetry

File: packages/dialog/src/commands/index.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { Command, flags } from '@oclif/command';
```
Comment (munozemilio) [inline packages/dialog/src/commands/index.ts]: use import {Command, flags} from '@microsoft/bf-cli-command'

File: packages/dialog/package.json [hunk 10]
```diff
@@ -4,23 +4,33 @@
...
     "@oclif/plugin-help": "^2.2.0",
     "@oclif/test": "^1.2.5",
     "@oclif/tslint": "^3.1.1",
-    "@types/chai": "^4.1.7",
+    "@types/chai": "^4.2.0",
+    "@types/fs-extra": "^8.0.0",
     "@types/mocha": "^5.2.7",
-    "@types/node": "^10.14.14",
+    "@types/node": "^10.14.15",
+    "ajv": "^6.9.1",
```
Comment (chrimc62) [inline packages/dialog/package.json]: Should be in dependencies.

File: packages/dialog/package.json (hunk 4 above)
Comment (tomlm) [reply packages/dialog/package.json]: I put it back

---
In reply to: [313947465](https://github.com/microsoft/botframework-cli/pull/88#discussion_r313947465) [](ancestors = 313947465)

**Pull request merged**